quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java

```diff
@@ -76,6 +80,19 @@ public Handler<RoutingContext> getManagementHandler() {
         return routingContext -> routingContext.response().end("Keycloak Management Interface");
     }
 
+    public void validatePersistenceUnits(List<String> descriptors) {
+        List<String> notSetPersistenceUnitsDBKinds = descriptors.stream()
+                .filter(descriptorName -> !descriptorName.equals(DEFAULT_PERSISTENCE_UNIT)) // not default persistence unit
+                // map PU name to DB kind option like: user-store -> db-kind-user-store
+                .map(descriptorName -> DatabaseOptions.getResultNamedKey(DatabaseOptions.DB, descriptorName).orElseThrow())
```

Comment:
If the persistence.xml specifies a datasource name, that may not match the persistence unit name.

In the vein of https://github.com/keycloak/keycloak/compare/main...shawkins:keycloak:pu we could default the datasource name to that of the persistence unit when not specified to cut down on what has to be in the xml.